### DISCUSSION TITLE
Does cargo auditable work with cargo capi?

### DISCUSSION THREAD

**Discussion by xnox** [rust-secure-code/cargo-auditable#175] [Q&A] [answered]
I was trying to build rustls-ffi C API library which uses capi.

I don't know if it works with audit, and how to extract the information later on.

How should it be invoked?

And how to extract audit data later on?

**Comment by Shnatsel** [comment]
> How should it be invoked?

`cargo auditable capi build` should work, in theory.

> And how to extract audit data later on?

This is documented in the README, [here](https://github.com/rust-secure-code/cargo-auditable?tab=readme-ov-file#is-there-any-tooling-to-consume-this-data) and [here](https://github.com/rust-secure-code/cargo-auditable/blob/master/PARSING.md). I suggest `cargo install rust-audit-info` as the simplest debugging tool.

> I don't know if it works with audit

I tried it and it doesn't seem to work. The culprit is this line:

https://github.com/rust-secure-code/cargo-auditable/blob/ceb4475d237b0296a3ddb946e0337fb658743ccc/cargo-auditable/src/rustc_wrapper.rs#L25

If I replace it with `if true`, it works. That's because `cargo capi` passes both `--print` and `--emit` arguments to `rustc`, which I didn't realize was possible.

**Reply by Shnatsel** [reply]
I've created an issue to track this: https://github.com/rust-secure-code/cargo-auditable/issues/176

**Reply by Shnatsel** [reply]
And a PR to fix this: https://github.com/rust-secure-code/cargo-auditable/pull/177

**Comment by Shnatsel** [comment, marked as answer]
I've released v0.6.6 that makes `cargo auditable capi build` work.
